401-code-challenges/Data-Structures/src/main/java/graph/Graph.java

```diff
@@ -50,4 +50,30 @@ public int size(){
         return nodes.size();
     }
 
+    public List<Integer> BFS(Node node){
+        List<Integer> order = new ArrayList<>();
+        boolean visited[] = new boolean[nodes.size()];
```

Comment:
You first `[]` are in the wrong place.This should be:
`boolean[] visited = new boolean[nodes.size()];`